controllers/provider-gcp/pkg/internal/infrastructure/terraform.go

```diff
@@ -144,8 +144,31 @@ func RenderTerraformerChart(
 	config *api.InfrastructureConfig,
 	cluster *extensionscontroller.Cluster,
 ) (*TerraformFiles, error) {
+	var (
+		aggregationInterval string
+		flowSampling        float32
+		metadata            string
+	)
+
 	values := ComputeTerraformerChartValues(infra, account, config, cluster)
 
+	if config.Networks.FlowLogs != nil {
+		if config.Networks.FlowLogs.AggregationInterval != nil {
+			aggregationInterval = *config.Networks.FlowLogs.AggregationInterval
+		}
+		if config.Networks.FlowLogs.FlowSampling != nil {
+			flowSampling = *config.Networks.FlowLogs.FlowSampling
+		}
+		if config.Networks.FlowLogs.Metadata != nil {
+			metadata = *config.Networks.FlowLogs.Metadata
+		}
+		values["networks"].(map[string]interface{})["flowLogs"] = map[string]interface{}{
+			"aggregationInterval": aggregationInterval,
+			"flowSampling":        flowSampling,
+			"metadata":            metadata,
+		}
+	}
```

Comment:
Instead, I'd suggest something like this:

```suggestion
	if flowLogs := config.Networks.FlowLogs; flowLogs != nil {
	  var flowLogs map[string]interface{}
	  
	  if flowLogs.AggregationInterval != nil {
	    flowLogs["aggregationInterval"] = *flowLogs.AggregationInterval
	  }
	  
	  if flowLogs.FlowSampling != nil {
	    flowLogs["flowSampling"] = *flowLogs.FlowSampling
	  }
	  
	  if flowLogs.Metadata != nil {
	    flowLogs["metadata"] = *flowLogs.Metadata
	  }
	  
	  values["networks"].(map[string]interface{})["flowLogs"] = flowLogs
	}
```